ui/layouts/route-with-layout.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ComponentType, ReactNode, useMemo } from 'react';
+import { Route, RouteComponentProps } from 'react-router-dom';
+import Authenticated from '../helpers/higher-order-components/authenticated/authenticated.container';
+
+type Props = {
+  path: string;
+  exact?: boolean;
+  component: ComponentType<RouteComponentProps>;
+  layout: ComponentType<{ children: ReactNode }>;
+  authenticated?: boolean;
+};
+
+/*
+Once migrated, we can use layouts like this:
+  <Route path="/settings" element={<SettingsLayout />}>
```

Comment:
A helper so that once we migrate to React Router v6 and use nested routes, it's easier to see where to group by common layout